From mcuelenaere@gmail.com  Mon Mar 15 16:33:42 2010
From: Maurus Cuelenaere <mcuelenaere@gmail.com>
Date: Mon, 8 Mar 2010 18:20:59 +0100
Subject: USB: gadget: add gadget controller number for s3c-hsotg driver
To: linux-usb@vger.kernel.org
Message-ID: <b2e8f9f71003080920q1e473a8an1d8c361082d2ff6b@mail.gmail.com>


This prevents some drivers from complaining that no bcdDevice id was set.

Signed-off-by: Maurus Cuelenaere <mcuelenaere@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/gadget_chips.h |    8 ++++++++
 1 file changed, 8 insertions(+)

--- a/drivers/usb/gadget/gadget_chips.h
+++ b/drivers/usb/gadget/gadget_chips.h
@@ -136,6 +136,12 @@
 #define	gadget_is_r8a66597(g)	0
 #endif
 
+#ifdef CONFIG_USB_S3C_HSOTG
+#define gadget_is_s3c_hsotg(g)    (!strcmp("s3c-hsotg", (g)->name))
+#else
+#define gadget_is_s3c_hsotg(g)    0
+#endif
+
 
 /**
  * usb_gadget_controller_number - support bcdDevice id convention
@@ -192,6 +198,8 @@ static inline int usb_gadget_controller_
 		return 0x24;
 	else if (gadget_is_r8a66597(gadget))
 		return 0x25;
+	else if (gadget_is_s3c_hsotg(gadget))
+		return 0x26;
 	return -ENOENT;
 }
 
